code/main/media/mediaModule.brs

```diff
@@ -0,0 +1,245 @@
+' * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+' * you may not use this file except in compliance with the License. You may obtain a copy
+' * of the License at http://www.apache.org/licenses/LICENSE-2.0
+' *
+' * Unless required by applicable law or agreed to in writing, software distributed under
+' * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+' * OF ANY KIND, either express or implied. See the License for the specific language
+' * governing permissions and limitations under the License.
+' *
+' *****************************************************************************************
+
+' *********************************** MODULE: Media ***************************************
+function _adb_isMediaModule(module as object) as boolean
+    return (module <> invalid and module.type = "com.adobe.module.media")
+end function
+
+function _adb_MediaModule(configurationModule as object, identityModule as object) as object
+    if _adb_isConfigurationModule(configurationModule) = false then
+        _adb_logError("_adb_MediaModule() - configurationModule is not valid.")
+        return invalid
+    end if
+
+    if _adb_isIdentityModule(identityModule) = false then
+        _adb_logError("_adb_EdgeModule() - identityModule is not valid.")
+        return invalid
+    end if
+
+    module = _adb_AdobeObject("com.adobe.module.media")
+
+    module.Append({
+        ' constants
+        _CONSTANTS: _adb_InternalConstants(),
+
+        ' external dependencies
+        _configurationModule: configurationModule,
+        _identityModule: identityModule,
+        _edgeRequestWorker: _adb_EdgeRequestWorker(),
+        _sessionManager: _adb_MediaSessionManager(),
+
+        ' Event data example:
+        ' {
+        '     clientSessionId: "xx-xxxx-xxxx",
+        '     tsObject: { ts_inISO8601: "2019-01-01T00:00:00.000Z", ts_inMillis: 1546300800000 },
+        '     xdmData: { xdm: {} }
+        '     configuration: { }
+        ' }
+        processEvent: sub(requestId as string, eventData as object)
+            mediaEventType = eventData.xdmData.xdm.eventType
+            clientSessionId = eventData.clientSessionId
+            tsObject = eventData.tsObject
+            xdmData = eventData.xdmData
+
+            if mediaEventType = m._CONSTANTS.MEDIA.SESSION_START_EVENT_TYPE
+                sessionConfig = eventData.configuration
+                m._sessionStart(clientSessionId, sessionConfig, xdmData, tsObject)
+                return
+            end if
+
+            if mediaEventType <> invalid and _adb_isStringInArray(mediaEventType, m._CONSTANTS.MEDIA.EVENT_TYPES)
+                m._actionInSession(requestId, clientSessionId, xdmData, tsObject)
+            else
+                _adb_logError("processEvent() - media event type is invalid: " + FormatJson(mediaEventType))
+            end if
+        end sub,
+
+        _mediaConfigIsNotReady: function() as boolean
```

Comment:
can we invert this? 
`_isMediaConfigReady()`

Reply:
It's used for checking the `bailout condition`. I can't see the obvious difference. I can update it, if you have a strong opinion. 
```
if not _isMediaConfigReady()
   return
end if
```

```
if _mediaConfigIsNotReady()
   return
end if
```